kolibri/plugins/coach/assets/src/app.js

```diff
@@ -139,9 +139,16 @@ class CoachToolsModule extends KolibriApp {
       }
 
       if (promises.length > 0) {
-        Promise.all(promises).then(next, error => {
-          this.store.dispatch('handleApiError', { error });
-        });
+        Promise.all(promises)
+          .catch(error => {
+            this.store.dispatch('handleApiError', { error });
+          })
+          .catch(() => {
+            // We catch here because `handleApiError` throws the error back again, in this case,
+            // we just want things to keep moving so that the AuthMessage shows as expected
+            next();
```

Comment:
This may end up calling next twice, because we catch and then then - but I don't think this matters.

Reply:
I think calling `next()` basically stops any further code execution so if one promise rejects at any point, we'll have `next` called and will be redirected altogether.

That said, I did look back at [Promise.all's docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all#promise.all_fail-fast_behavior) and see that in this case, the `then` would be called if `next` wasn't called because of the chaining order.

If the `then` comes first in the chain series, then it would not be called if the `catch` blocks ran because Promise.all handles any rejection immediately.  But, if  you `catch` errors and chain a `then` to the `catch` though, the `then` is run.

Might be misinterpreting a bit there but seemed to be the case when I played around a bit w/ Promises in the terminal

In any case, I think you're right that it won't have any effect as-is.